libs/execution_context/ebpf_program.c

```diff
@@ -976,23 +976,30 @@ ebpf_program_set_tail_call(_In_ const ebpf_program_t* next_program)
 }
 
 void
-ebpf_program_invoke(_In_ const ebpf_program_t* program, _Inout_ void* context, _Out_ uint32_t* result)
+ebpf_program_invoke(_Inout_ ebpf_program_t* program, _Inout_ void* context, _Out_ uint32_t* result)
 {
     // High volume call - Skip entry/exit logging.
     ebpf_program_tail_call_state_t state = {0};
     const ebpf_program_t* current_program = program;
 
-    if (!program || program->invalidated) {
+    bool rundown_protection_acquired = false;
+    bool program_state_stored = false;
+
+    if (!program || !ebpf_rundown_protection_acquire(&program->rundown.protection)) {
```

Comment:
I'd recommend renaming ebpf_rundown_protection_acquire.  The point is just to take a reference, and if we replace the blocking rundown protection with an async implementation, we shouldn't have to touch this code.  Currently the name implies a specific blocking implementation.  Same goes with renaming the variable in 985, 1053, etc.